src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -418,6 +425,84 @@ public void deleteTemplate(String projectId, String templateName) throws IOExcep
         return instanceTemplates;
     }
 
+    /**
+     * Creates persistent disk snapshot for Compute Engine instance.
+     * This method blocks until the operation completes.
+     *
+     * @param projectId Google cloud project id (e.g. my-project).
+     * @param zone Instance's zone.
+     * @param instanceId Name of the instance whose disks to take a snapshot of.
+     *
+     * @throws IOException If an error occured in snapshot creation.
+     * @throws InterruptedException If snapshot creation is interrupted.
+     */
+    public void createSnapshot(String projectId, String zone, String instanceId) throws IOException, InterruptedException {
+        try {
+            zone = nameFromSelfLink(zone);
+            Instance instance = compute.instances().get(projectId, zone, instanceId).execute();
+
+            for (AttachedDisk disk : instance.getDisks()) {
+                    String diskId = nameFromSelfLink(disk.getSource());
+                    createSnapshotForDisk(projectId, zone, diskId);
```

Comment:
Since this method blocks until snapshot is done, it'd be nice to parallelize this. Mind adding a TODO comment here to improve in a later commit?

Reply:
done!